Add samples for csv<->table conversions

Manuri opened the pull request:
## Purpose
1. Add sample for reading a CSV file as a table
2. Ass sample for writing a table to a CSV file

Resolves https://github.com/ballerina-platform/ballerina-examples/issues/788

Manuri committed:
Add samples for csv<->table conversions

anupama-pathirage commented on examples/csv-io/csv_io.bal:
```diff
@@ -1,6 +1,17 @@
+    string id,
+    float salary,
+};
+
```
This sample is assuming user has sample.csv in the ../files location. Instead of that shall we re-organize this to create the .csv file first based on some data( may be using some record data in inmemory table) and then read that .csv file. In that way user can run the sample without having any per-requisists.

anupama-pathirage commented on examples/csv-io/csv_io.bal:
```diff
@@ -46,15 +57,65 @@ function main(string... args) {
+    // Read the `.CSV` file as a table.
+    csvChannel = io:openCsvFile(srcFileName);
+    table<Salary> salaryTable;
+    try {
```
Why do we need a try catch here? getTable method is returning errors

Manuri replied:
Yeah I needed to confirm this. Wasn't really sure if there were cases that throw un-recoverable errors in I/O operations.

anupama-pathirage commented on examples/csv-io/csv_io.bal:
```diff
@@ -46,15 +57,65 @@ function main(string... args) {
+    string targetFileName = "./files/output.csv";
+    // Opening CSV channel in "write" mode.
+    csvChannel = io:openCsvFile(targetFileName, mode = "w");
+    try {
```
Why do we need try catch here?

Manuri commented on examples/csv-io/csv_io.bal:
```diff
@@ -46,15 +57,65 @@ function main(string... args) {
+        match csvChannel.getTable(Employee) {
+            table<Employee> employeeTable => {
+                io:println("Creating `salaryTable` using `select` operation");
+                salaryTable = employeeTable.select(getSalary);
```
Well we can simply print the data here. I used select so that I can create a table out of the data obtained from the file which could then be written to another file, instead of writing the same table back.

anupama-pathirage replied:
IMO having select here gives the impression that we can't directly use the returned table<Employee> employeeTable. We can use that table in any way user want, not just select

Manuri committed:
Refactor to make the sample easily understandable

Here, at the beginning data is inserted to the sample file to make user have least pre-requisites. Also removed 'select' functionality to avoid to expose least unrelated functionality
